### ISSUE TITLE
kadmin: pw-expiration-time cannot bet set when adding new principal with random key

### ISSUE THREAD

**Issue by klada** [heimdal/heimdal#360]
When creating new principals with a random key the `pw-expiration-time` argument is ignored. The value is always set to the default of 1 year, even if the principal expiration time is shorter.

Example:
```
kadmin -l add \
  --random-key \
  --max-ticket-life=30h \
  --max-renewable-life=30d \
  --expiration-time=2018-12-31 \
  --pw-expiration-time=2018-12-31 \
  --use-defaults \
  foobar
```

Result:

```
kadmin -l get foobar
            Principal: foobar@REALM.LAN
    Principal expires: 2018-12-31 23:59:59 UTC
     Password expires: 2019-02-14 15:03:08 UTC
 Last password change: 2018-02-14 15:03:08 UTC
```

**Comment by lhoward** [comment]
@klada please try fix in pull request above

**Comment by lhoward** [comment]
Merged in def6e4fc

**Comment by lhoward** [comment]
Great. I can’t even remember this fix now, it’s clearly been a busy two weeks!
